components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -827,6 +827,14 @@ void BraveProfileSyncServiceImpl::FetchSyncRecords(const bool bookmarks,
     category_names.push_back(kPreferences);  // "PREFERENCES";
   }
 
+  base::Time last_compact_time = brave_sync_prefs_->GetLastCompactTime();
+  if (tools::IsTimeEmpty(last_compact_time) ||
+      base::Time::Now() - last_compact_time >
+          base::TimeDelta::FromDays(kCompactPeriodInDays)) {
+    brave_sync_client_->SendCompactSyncCategory(kBookmarks);
+    brave_sync_prefs_->SetLastCompactTime(base::Time::Now());
```

Comment:
shouldn't we wait to do this until the compaction is complete?

Reply:
You mean, when the objects are really deleted on s3?
The compaction API is designed to have no callback, if we want callback when the object is really deleted, we would need extra listObject query to check. The deletion is not instant, it takes approximately 1 minute

Reply:
Also, the compaction could take a significant long time to pull down all the record if the recordset is huge

Reply:
addressed in bb0c6c3fda0de8544d383a2fc09f8f59e748ca83
which requires deps update when https://github.com/brave/sync/pull/355 is merged